core/formats/formats_10.cpp

```diff
@@ -3316,40 +3492,58 @@ class postings_reader final : public postings_reader_base {
 
   irs::doc_iterator::ptr wanderator(
     IndexFeatures field_features, IndexFeatures required_features,
-    const term_meta& meta,
-    [[maybe_unused]] const WanderatorOptions& options) final {
+    const term_meta& meta, [[maybe_unused]] const WanderatorOptions& options,
+    WandContext ctx, WandInfo info) final {
     if constexpr (FormatTraits::wand()) {
-      if (meta.docs_count <= FormatTraits::block_size() ||
-          IndexFeatures::NONE == (field_features & IndexFeatures::FREQ)) {
+      const auto* scorer = [&]() -> const Scorer* {
+        if (meta.docs_count <= FormatTraits::block_size() ||
+            info.mapped_index == irs::WandContext::kDisable ||
+            scorers_.size() <= ctx.index) {
+          return nullptr;
+        }
+
+        const auto& scorer = *scorers_[ctx.index];
+        const auto scorer_features = scorer.index_features();
+
+        if (scorer_features != (field_features & scorer_features)) {
+          return nullptr;
+        }
+
+        return &scorer;
+      }();
+
+      if (!scorer) {
+        // FIXME(gnusi): wand simulation
```

Comment:
I'm not sure but I thought it will be not here, but in default seek2block implementation?